tests/models/test_boundary_condition.py

```diff
@@ -113,57 +113,97 @@ def test_boundary_condition_mixin(t_end: int):
 """Here follows mixins related to testing of Robin limit cases, and eventually the test itself. """
 
 
-class BCValues:
-    """Set boundary values for momentum balance and mass and energy balance."""
+class BCValuesLeftSide:
+    """Boundary values for primary variables on west and east boundary.
+
+    Used for:
+    * Momentum balance
+    * Mass and energy balance.
+
+    """
+
+    def not_dir_inds(self, g):
+        """Indices for the non-Dirichlet boundaries for test.
+
+        The Robin limit case test tests setups with Dirichlet on west/east boundaries,
+        and Robin approximating Dirichlet on the north/south boundaries. This method
+        returns the indices of the north and south boundaries.
+
+        """
+        bounds = self.domain_boundary_sides(g)
+        return bounds.north + bounds.south
+
+    def dir_inds(self, g):
+        """Indices for the Dirichlet boundaries for test.
+
+        The Robin limit case test tests setups with Dirichlet on west/east boundaries,
+        and Robin approximating Dirichlet on the north/south boundaries. This method
+        returns the indices of the east and west boundaries.
+
+        """
+        bounds = self.domain_boundary_sides(g)
+        return bounds.west + bounds.east
 
     def bc_values_displacement(self, bg: pp.BoundaryGrid) -> np.ndarray:
-        """Assigns displacement values in the x-direction of the west boundary."""
-        values = np.ones((self.nd, bg.num_cells))
-        bounds = self.domain_boundary_sides(bg)
-        values[0][bounds.west] += np.ones(len(values[0][bounds.west]))
+        """Assigns displacement values in the x-direction of the west and east
```

Comment:
Since the dir_inds method may be overridden, I suggest:
```suggestion
        """Assigns displacement values in the x-direction of the Dirichlet
```
If you agree, please roll out throughout the file.